### PR TITLE
[INLONG-9820][Dashboard][Manager] Update Pulsar source field for Ingestion

### LINKED ISSUE #9820: [Improve][Dashboard] Update Pulsar source field
### Description

Update Pulsar source field for `Ingestion`

### InLong Component

InLong Dashboard

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haifxu**
### Prepare a Pull Request

- Fixes #9820 

### Motivation

1. Dashboard add fields `subscription`, `scanStartupMode`, `subscriptionType`, `resetTime` and `dataTimeZone` for Ingestion.
2. Add fields `subscriptionType` and `resetTime` in Manager.
3. Change `resetTime` field type and `subscriptionPosition` name for better interaction with the manager.

### Modifications

image

### CODE REVIEW

File: inlong-dashboard/src/plugins/sources/defaults/Pulsar.ts [hunk 1]
```diff
@@ -123,13 +125,87 @@ export default class PulsarSource
...
+  })
+  @ColumnDecorator()
+  @IngestionField()
+  @I18n('meta.Sources.Pulsar.SubscriptionPosition')
+  scanStartupMode: string;
+
+  @FieldDecorator({
+    type: 'select',
+    initialValue: 'Shared',
+    tooltip: i18n.t('meta.Sources.Pulsar.SubscriptionTypeHelp'),
```
Comment (dockerzhang) [inline inlong-dashboard/src/plugins/sources/defaults/Pulsar.ts:163]: the subscription type should be configured by the user.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/pulsar/PulsarSourceRequest.java [hunk 2]
```diff
@@ -76,6 +76,12 @@ public class PulsarSourceRequest extends SourceRequest {
             + " Available options are earliest, latest, external-subscription, and specific-offsets.")
     private String scanStartupMode = "earliest";
 
+    @ApiModelProperty("Configure the Pulsar subscription mode. Available options are Exclusive, Shared, and Failover.")
```
Comment (aloyszhang) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/pulsar/PulsarSourceRequest.java:79]: What about the Key_Shared subscription type?

File: inlong-dashboard/src/plugins/sources/defaults/Pulsar.ts [hunk 3]
```diff
@@ -123,13 +125,87 @@ export default class PulsarSource
...
+    props: values => ({
+      disabled: values?.status === 101,
+      options: [
```
Comment (aloyszhang) [inline inlong-dashboard/src/plugins/sources/defaults/Pulsar.ts:166]: Why not support the Key_Shared type?

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/pulsar/PulsarSourceRequest.java (hunk 2 above)
Comment (haifxu) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/pulsar/PulsarSourceRequest.java]: We may no longer need users to specify subscription types.

File: inlong-dashboard/src/plugins/sources/defaults/Pulsar.ts [hunk 4]
```diff
@@ -21,6 +21,8 @@ import { DataWithBackend } from '@/plugins/DataWithBackend';
 import { RenderRow } from '@/plugins/RenderRow';
 import { RenderList } from '@/plugins/RenderList';
 import { SourceInfo } from '../common/SourceInfo';
+import dayjs from 'dayjs';
```
[the remark this answers is not in the document]
Comment (haifxu) [reply inlong-dashboard/src/plugins/sources/defaults/Pulsar.ts:24]: Now use dayjs to convert the time into a timestamp.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/pulsar/PulsarSourceRequest.java [hunk 5]
```diff
@@ -76,6 +76,9 @@ public class PulsarSourceRequest extends SourceRequest {
             + " Available options are earliest, latest, external-subscription, and specific-offsets.")
     private String scanStartupMode = "earliest";
 
+    @ApiModelProperty("Reset consumption time")
```
Comment (aloyszhang) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/pulsar/PulsarSourceRequest.java:79]:
```suggestion
    @ApiModelProperty("Reset subscription time")
```

File: inlong-dashboard/src/ui/locales/en.json [hunk 6]
```diff
@@ -131,6 +131,11 @@
   "meta.Sources.Iceberg.Warehouse": "Warehouse",
   "meta.Sources.Pulsar.PulsarTenant": "Pulsar tenant",
   "meta.Sources.Pulsar.Namespace": "Namespace",
+  "meta.Sources.Pulsar.Subscription": "Subscription",
+  "meta.Sources.Pulsar.SubscriptionHelp": "If not specified, the default is: inlong-agent-{inlongStreamId}",
+  "meta.Sources.Pulsar.SubscriptionPosition": "Subscription Position",
+  "meta.Sources.Pulsar.ResetTime": "Reset consumption time",
```
Comment (aloyszhang) [inline inlong-dashboard/src/ui/locales/en.json:137]:
```suggestion
  "meta.Sources.Pulsar.ResetTime": "Reset subscription time",
```

**Review by justinwwhuang — DISMISSED** [review]
LGTM

File: inlong-dashboard/src/ui/locales/cn.json [hunk 7]
```diff
@@ -131,6 +131,11 @@
   "meta.Sources.Iceberg.Warehouse": "仓库路径",
   "meta.Sources.Pulsar.PulsarTenant": "Pulsar 租户",
   "meta.Sources.Pulsar.Namespace": "命名空间",
+  "meta.Sources.Pulsar.Subscription": "Pulsar 订阅",
+  "meta.Sources.Pulsar.SubscriptionHelp": "若不指定则默认为：inlong-agent-{inlongStreamId}",
+  "meta.Sources.Pulsar.SubscriptionPosition": "订阅位点",
+  "meta.Sources.Pulsar.ResetTime": "重置消费时间",
+  "meta.Sources.Pulsar.ResetTimeHelp": "从当前时间开始消费，若不指定则从 pulsar 已提交的位点消费",
```
Comment (dockerzhang) [inline inlong-dashboard/src/ui/locales/cn.json:138]:
```suggestion
  "meta.Sources.Pulsar.ResetTimeHelp": "从指定时间开始消费",
```

File: inlong-dashboard/src/ui/locales/en.json [hunk 8]
```diff
@@ -131,6 +131,11 @@
   "meta.Sources.Iceberg.Warehouse": "Warehouse",
   "meta.Sources.Pulsar.PulsarTenant": "Pulsar tenant",
   "meta.Sources.Pulsar.Namespace": "Namespace",
+  "meta.Sources.Pulsar.Subscription": "Subscription",
+  "meta.Sources.Pulsar.SubscriptionHelp": "If not specified, the default is: inlong-agent-{inlongStreamId}",
+  "meta.Sources.Pulsar.SubscriptionPosition": "Subscription Position",
+  "meta.Sources.Pulsar.ResetTime": "Reset subscription time",
+  "meta.Sources.Pulsar.ResetTimeHelp": "Reset the subscription associated with this consumer to a specific message publish time.",
```
Comment (dockerzhang) [inline inlong-dashboard/src/ui/locales/en.json:138]:
```suggestion
  "meta.Sources.Pulsar.ResetTimeHelp": "Reset the subscription time.",
```

File: inlong-dashboard/src/ui/locales/en.json [hunk 9]
```diff
@@ -131,6 +131,11 @@
   "meta.Sources.Iceberg.Warehouse": "Warehouse",
   "meta.Sources.Pulsar.PulsarTenant": "Pulsar tenant",
   "meta.Sources.Pulsar.Namespace": "Namespace",
+  "meta.Sources.Pulsar.Subscription": "Subscription",
+  "meta.Sources.Pulsar.SubscriptionHelp": "If not specified, the default is: inlong-agent-{inlongStreamId}",
+  "meta.Sources.Pulsar.SubscriptionPosition": "Subscription Position",
+  "meta.Sources.Pulsar.ResetTime": "Reset subscription time",
+  "meta.Sources.Pulsar.ResetTimeHelp": "Reset the subscription time.",
```
Comment (dockerzhang) [inline inlong-dashboard/src/ui/locales/en.json:138]:
```suggestion
  "meta.Sources.Pulsar.ResetTimeHelp": "Reset the subscription time",
```

**Review by dockerzhang — APPROVED** [review]

File: inlong-dashboard/src/plugins/sources/common/SourceDefaultInfo.ts [hunk 10]
```diff
@@ -146,10 +147,16 @@ export class SourceDefaultInfo implements DataWithBackend, RenderRow, RenderList
   readonly modifier: string;
 
   parse(data) {
+    if (data.resetTime !== undefined) {
+      data.resetTime = dayjs(data.resetTime).format('YYYY-MM-DD HH:mm:ss');
```
Comment (leezng) [inline inlong-dashboard/src/plugins/sources/common/SourceDefaultInfo.ts:151]: Is resetTime a parameter that all Source plugins have? If not, do not add it to default.

**Review by dockerzhang — APPROVED** [review]

**Review by leezng — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**